src/goto-programs/goto_instruction_code.h

```diff
@@ -537,4 +538,18 @@ inline code_returnt &to_code_return(codet &code)
   return static_cast<code_returnt &>(code);
 }
 
+/// \brief Builds a \ref code_function_callt to `__CPROVER_havoc_slice(p, size)`.
+/// \param p the pointer argument
+/// \param size the size argument
+/// \return a \ref code_function_callt expression `nil_exprt() := __CPROVER_havoc_slice(p, size)`
+inline code_function_callt havoc_slice_call(exprt &p, exprt &size)
+{
+  code_function_callt::argumentst arguments;
+  arguments.push_back(p);
+  arguments.push_back(size);
+  symbol_exprt havoc_slice_function =
+    symbol_exprt{CPROVER_PREFIX "havoc_slice", empty_typet()};
```

Comment:
This needs to be of a `code_typet` type, and then makes me wonder whether you should rather fetch this symbol from the symbol table. And that, in turn, makes me wonder whether such a function should make sure it is present in the symbol table.

Reply:
The symbol's signature is only declared in `src/ansi-c/cprover_builtin_headers.h`.
If someone were to build a goto model programatically without using the C front-end, is there a guarantee that this signature will be present in the symbol table ?

Reply:
No, there is no such guarantee. Hence, it would be necessary to check, and add the symbol if it isn't present just yet.